opentracing-cdi-example/src/main/java/io/opentracing/contrib/cdi/example/SpanContextProducer.java

```diff
@@ -0,0 +1,18 @@
+package io.opentracing.contrib.cdi.example;
+
+import io.opentracing.SpanContext;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+public class SpanContextProducer {
+    @Inject
+    HttpServletRequest request;
+
+    @Produces
+    public SpanContext produceSpanContext() {
+        return (SpanContext) request.getAttribute(TracingFilter.SERVER_SPAN_CONTEXT);
```

Comment:
Availability of an _active_ span basically means:
Here's some current, ongoing span. If you want to create a span by start / finish, it may be beneficial to become _child of_ this ongoing span.

I would definitely _not_ get involved with the lifecycle of this parent!